raft.go

```diff
@@ -757,7 +757,10 @@ func (r *raft) reset(term uint64) {
 }
 
 func (r *raft) appendEntry(es ...pb.Entry) (accepted bool) {
-	li := r.raftLog.lastIndex()
+	li, lt := r.raftLog.tip()
+	if r.Term < lt {
+		r.logger.Panicf("%x appending out-of-order term: %d < %d", r.id, r.Term, lt)
```

Comment:
nit: you aren't checking Terms in `es`, so msg is a bit confusing. Maybe `appending out-of-order term` -> `last raft log entry has term greater then raft struct`

Reply:
how about using `warning` instead of `panic`?